File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t):
            return False

        sDict = {}
        tDict = {}

        for x in s:
            sDict[x] = sDict.get(x, 0) + 1
        
        for x in t:
            tDict[x] = tDict.get(x, 0) + 1

        return sDict == tDict
        

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        // s = "anagram"
        // t = "nagaram"

        ////// Solution 1: sorting

        // if(s.size() != t.size()){
        //     return false;
        // }

        // sort(s.begin(), s.end());
        // sort(t.begin(), t.end());

        // return s == t;


        if(s.size() != t.size()){
            return false;
        }

        unordered_map<char, int> sMap;
        unordered_map<char, int> tMap;

        for(int i = 0; i < s.size(); i++){
            sMap[s[i]]++;
            tMap[t[i]]++;
        }

        return sMap == tMap;


    }
};